### ISSUE TITLE
python36Packages setuptools missing dependency, won't build?

### ISSUE THREAD

**Issue by dtzWill** [NixOS/nixpkgs#23406]
## Issue description

Looks like setuptools won't build on python 3.6, which blocks building many other packages.

### Steps to reproduce

On 17.03 branch:

```
$ nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion
"17.03beta143.fba286e"
$ nix-shell -p python35Packages.isodate
[nix-shell] # works
$ nix-shell -p python36Packages.isodate --no-build-hook                                                                                      15:32:20 on 17-03-02
these derivations will be built:
  /nix/store/qm1m58g6hfp0zls4y3zpjnh3qjaznrgk-python3.6-setuptools-30.2.0.drv
  /nix/store/w3i1f6a9pwl5sczk4rkfndgm211kwbcv-python3.6-isodate-0.5.4.drv
building path(s) ‘/nix/store/np7ijz6sr93q3a59clr7l83dlxc29km5-python3.6-setuptools-30.2.0’
unpacking sources
unpacking source archive /nix/store/70xydxh5ik64fc3r7rswhqqgly512n26-setuptools-30.2.0.tar.gz
source root is setuptools-30.2.0
setting SOURCE_DATE_EPOCH to timestamp 1480866953 of file setuptools-30.2.0/setuptools.egg-info/top_level.txt
patching sources
configuring
no configure script, doing nothing
building
no Makefile, doing nothing
installing
Traceback (most recent call last):
  File "setup.py", line 11, in <module>
    import setuptools
  File "/tmp/nix-build-python3.6-setuptools-30.2.0.drv-0/setuptools-30.2.0/setuptools/__init__.py", line 10, in <module>
    from setuptools.extern.six.moves import filter, filterfalse, map
  File "/tmp/nix-build-python3.6-setuptools-30.2.0.drv-0/setuptools-30.2.0/setuptools/extern/__init__.py", line 1, in <module>
    from pkg_resources.extern import VendorImporter
  File "/tmp/nix-build-python3.6-setuptools-30.2.0.drv-0/setuptools-30.2.0/pkg_resources/__init__.py", line 35, in <module>
    import plistlib
  File "/nix/store/sh3dirnxw1gl3yb9fvi02sxzl3cp7a5m-python3-3.6.0/lib/python3.6/plistlib.py", line 65, in <module>
    from xml.parsers.expat import ParserCreate
  File "/nix/store/sh3dirnxw1gl3yb9fvi02sxzl3cp7a5m-python3-3.6.0/lib/python3.6/xml/parsers/expat.py", line 4, in <module>
    from pyexpat import *
ModuleNotFoundError: No module named 'pyexpat'
builder for ‘/nix/store/qm1m58g6hfp0zls4y3zpjnh3qjaznrgk-python3.6-setuptools-30.2.0.drv’ failed with exit code 1
cannot build derivation ‘/nix/store/w3i1f6a9pwl5sczk4rkfndgm211kwbcv-python3.6-isodate-0.5.4.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/w3i1f6a9pwl5sczk4rkfndgm211kwbcv-python3.6-isodate-0.5.4.drv’ failed
/run/current-system/sw/bin/nix-shell: failed to build all dependencies
```

Same on current nixpkgs master, output omitted but available if needed.

## Technical details

* System: (NixOS: `nixos-version`, Ubuntu/Fedora: `lsb_release -a`, ...)
* Nix version: (run `nix-env --version`)
* Nixpkgs version: (run `nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion`)

**Comment by Mic92** [comment]
related to https://github.com/NixOS/nixpkgs/issues/23325

**Comment by FRidh** [comment]
A fix (a1f6b8b) is in staging.

**Comment by dtzWill** [comment]
Thanks!
